app/models/chargeback.rb

```diff
@@ -182,7 +182,10 @@ def self.report_cb_model(model)
     model.gsub(/^Chargeback/, "")
   end
 
+  # @return true if this class name is a subclass of chargeback?
+  # looking up the subclasses takes too long (50ms)
+  # so checking that the class name starts with "Chargeback" (10 is length of that string)
   def self.db_is_chargeback?(db)
-    subclasses.collect(&:to_s).include?(db)
+    db && db.length > 10 && db.starts_with?("Chargeback")
```

Comment:
@jrafanie 

> why do we want to check if this a chargeback subclass?

We can have report based on specialized classes ChargebackVm or ChargebackContainerProject and also we have some code  (mostly) on formatting and report components which should be run only for Chargeback report. Basically we are asking of miq_report record, "Are you Chargeback Report" ?

there are the places:
https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_report/formatting.rb#Lhttps://github.com/ManageIQ/manageiq/blob/master/app/models/miq_report/fo#L10#L100

https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_report/generator/html.rb#L54
and 
https://github.com/ManageIQ/manageiq/blob/master/app/controllers/report_controller/reports.rb#L394

maybe we can find out if these codes (under if leg self.db_is_chargeback?(db) ) can moved on different place ?

@kbrock what is the scenario  where we can see this perfomance issue ?
I can see that `db_is_chargeback` is called only when you are generating report ( and in definition of chargeback report) but I guess that your example is when  open page in meno Setting->Configuration, is the scenario ?